skiko/src/jvmMain/kotlin/org/jetbrains/skiko/cleanup.kt

```diff
@@ -0,0 +1,47 @@
+@file:OptIn(ExperimentalCoroutinesApi::class)
+
+package org.jetbrains.skiko
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import java.nio.file.Path
+import java.time.Duration
+import java.time.Instant
+import kotlin.concurrent.thread
+import kotlin.io.path.*
+
+/**
+ * Will asynchronously cleanup 'stale' entries in the [SkikoProperties.dataPath]:
+ * If directories are older than [SkikoProperties.dataCleanupDays], then they will be deleted.
+ */
+internal fun launchSkikoDataDirCleanupIfNecessary(dataDir: Path) {
+    thread(
+        name = ".skiko cleanup",
+        priority = Thread.MIN_PRIORITY
+    ) {
+        try {
+            withDirectoryLock(dataDir) {
+                doCleanupSkikoDataDir()
+            }
+        } catch (t: Throwable) {
+            Logger.error(t) { "Exception occurred during .skiko cleanup" }
+        }
+    }
+}
+
+private fun doCleanupSkikoDataDir() {
+    val dir = Path(SkikoProperties.dataPath)
+
+    dir.listDirectoryEntries().forEach { entry ->
```

Comment:
Another processes can also run cleaning and remove folders from this list.

But it is probably fine to just have an exception that is caught above. Let's just write a comment about this case, and that it is an expected case.

Reply:
No, another process cannot capture the .skiko/.lock file at the same time and remove entries. Am I missing something?

Reply:
You mean two processes running the cleaning?
This cannot happen, as we capture the lock in process 1 and check for existence once acquired. 
The root folder, which we iterate upon, is not deleted